### DISCUSSION TITLE
Issue with install Ubuntu 20.04.4 - sudo: command not found

### DISCUSSION THREAD

**Discussion by ChrisBomar87** [Nimdy/Dedicated_Valheim_Server_Script#285] [Q&A] [answered]
Good morning, 

I am having an issue after clicking install as seen here. It is showing several lines where "sudo: command not found"

It also will not let me perform the sudo command for chmod but will allow me to just chmod as I am signed in as root.

Any ideas what I am doing wrong? I'm new to ssh and terminal stuff.

Thanks in advance!

------

Thank you for using the Njord Menu system.
This appears to be the first time the menu has
been run on this system.
Installing the first Valheim server started.
ubuntu
20.04.4 LTS (Focal Fossa)
2
Checking for upgrades
./njordmenu.sh: line 584: sudo: command not found
Done
Install Git, Locate, Curl, Unzip and Net-Tools
./njordmenu.sh: line 606: sudo: command not found
Done
Installing software-properties-common package
./njordmenu.sh: line 628: sudo: command not found
Done
Adding multiverse REPO
./njordmenu.sh: line 638: sudo: command not found
Done
Adding i386 architecture
./njordmenu.sh: line 668: sudo: command not found
Done
Checking and updating system again
./njordmenu.sh: line 678: sudo: command not found
Done
Thanks for downloading the script, let's get started.
The following information is required for configuration files.
Read each step carefully.

A printout of data entered will be displayed to you
A non-root account will be created to run Valheim Server

This account is named steam and will act as a service account only
------------------------------------------------------------
----------------NONROOT STEAM ACCOUNT PASSWORD--------------
------------------------------------------------------------
Password must be 6 Characters or more
At least one number, one uppercase letter and one lowercase letter
------------------------------------------------------------
Good Example: Viking12
Bad Example: Vik!
------------------------------------------------------------

![SSH capture 1](https://user-images.githubusercontent.com/60636242/161201598-24ffa916-452a-4a0f-83ce-9967f999cff1.png)

**Comment by RedKrieg** [comment]
The lines with errors are trying to execute apt/dpkg commands and not finding them.  Are you running some custom ubuntu derivative that has replaced apt with some other package manager?  In a default Ubuntu install these are critical components and cannot be safely removed.  The only other thing I can think of would be a broken environment that doesn't include /usr/bin in the $PATH.

**Comment by ChrisBomar87** [comment]
I probably should have mentioned I’m running the official Ubuntu release in docker. No changes. Just a fresh container environment.

**Comment by RedKrieg** [comment, marked as answer]
Your docker container is not a complete OS install, it's missing components like the package manager.  If you want to install packages inside a docker container, you'll have to consult the docker documentation for info on installing apt/dpkg and any other applications that are missing in your container and used by the script.  You can run scripts like this to see the actual commands being attempted (warning, very verbose): `bash -x ./njordmenu.sh`

This project might be a better starting point for you if you're focused on learning to use docker to host Valheim: https://github.com/lloesche/valheim-server-docker

**Reply by ChrisBomar87** [reply]
I got that one working fine but like the server manager this one has. Is it possible to integrate it in the other suggested project?

**Reply by Nimdy** [reply]
Thanks @RedKrieg 

@Enigma619 

You could do it yourself but it might take some time.  If you have some experience with Docker, then you know it can be a pain in the butt with a lot of trial and error.

lloesche did a lot of hard work to get that running.  You might be able to get it working with Njord Menu... but mount points and persistent data structures might have issues... I only dabble with Docker, when needed.

If you figure it out and you want to share, please do... There might be others who wanted to do the same thing and you can save them tons of hours of face smashing.

Looking forward to see what transpires.
